### ISSUE TITLE
Polars `to_torch` produced wrong dimension when used with PyTorch `DataLoader`

### ISSUE THREAD

**Issue by AndhikaWB** [pola-rs/polars#20336]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

Direct PyTorch tensor (without Polars):

```python
import polars as pl
import torch
from torch.utils.data import Dataset, DataLoader

input = torch.tensor([[1, 2, 4, 5], [4, 3, 2, 9], [4, 3, 2, 9]])
print('Raw data:', input.shape)

class CustomDataset(Dataset):
    def __init__(self, input):
        self.input = input

    def __getitem__(self, idx):
        return self.input[idx], self.input[idx]

    def __len__(self):
        return self.input.shape[0]

dataset = CustomDataset(input)
print('Dataset:', dataset[:][0].shape)
temp = dataset[:][0]

loader = DataLoader(dataset = dataset, batch_size = 2)

for batch in loader:
    print('Dataloader:', batch[0].shape)
    break
```

```
Raw data: torch.Size([3, 4])
Dataset: torch.Size([3, 4])
Dataloader: torch.Size([2, 4])
```

Tensor from Polars dataframe (`to_torch`):

```python
df = pl.DataFrame({
    'a': [1, 4, 4],
    'b': [2, 3, 3],
    'c': [4, 2, 2],
    'd': [5, 9, 9]
})
print('Raw data:', df.shape)

class CustomDataset(Dataset):
    def __init__(self, df):
        self.df = df

    def __getitem__(self, idx):
        return self.df[idx].to_torch(), self.df[idx].to_torch()

    def __len__(self):
        return self.df.shape[0]

dataset = CustomDataset(df)
print('Dataset:', dataset[:][0].shape)

# Compare tensor with the previous code
print('Is equal:', torch.equal(dataset[:][0], temp))

loader = DataLoader(dataset = dataset, batch_size = 2)

for batch in loader:
    print('Dataloader:', batch[0].shape)
    break
```

```
Raw data: (3, 4)
Dataset: torch.Size([3, 4])
Is equal: True
Dataloader: torch.Size([2, 1, 4])
```

### Issue description

When using PyTorch custom `Dataset` where it makes use of Polars `to_torch` function, PyTorch `DataLoader` will produce 1 extra dimension:
- `torch.Size([2, 1, 4])` (from Polars), versus
- `torch.Size([2, 4])` (expected)

The dataset itself seems equal (see the code output), but the fact that `DataLoader` produce different result means there is actually a difference between the two

### Expected behavior

`to_torch` function should produce the same shape (`torch.Size([2, 4])`) when used with `DataLoader`, like when I use PyTorch tensor directly

### Installed versions

<details>

```
--------Version info---------
Polars:              1.17.1
Index type:          UInt32
Platform:            Windows-11-10.0.26100-SP0
Python:              3.12.7 | packaged by Anaconda, Inc. | (main, Oct  4 2024, 13:17:27) [MSC v.1929 64 bit (AMD64)]
LTS CPU:             False

----Optional dependencies----
adbc_driver_manager  <not installed>
altair               <not installed>
boto3                <not installed>
cloudpickle          <not installed>
connectorx           <not installed>
deltalake            <not installed>
fastexcel            <not installed>
fsspec               2024.2.0
gevent               <not installed>
google.auth          <not installed>
great_tables         <not installed>
matplotlib           3.9.2
nest_asyncio         1.6.0
numpy                1.26.3
openpyxl             <not installed>
pandas               2.2.3
pyarrow              18.0.0
pydantic             <not installed>
pyiceberg            <not installed>
sqlalchemy           <not installed>
torch                2.4.1+cu124
xlsx2csv             <not installed>
xlsxwriter           <not installed>
```

</details>

**Comment by alexander-beedie** [comment]
This isn't a bug, it's just an artefact of how you're using `to_torch()` here; when you index a Polars DataFrame row-wise (eg: `df[0]`) the result is _still_ a 2D DataFrame (composed of one row), and _not_ a 1D Series. Consequently `self.df[idx].to_torch()` returns `tensor([[1, 2, 4, 5]])` and not `tensor([1, 2, 4, 5])`; this where your extra dimension manifests 😉 

Note that we offer a dedicated `PolarsDataset` class that is drop-in compatible with `TensorDataset`, which you can either return with `df.to_torch("dataset")`, or import directly via `from polars.ml.torch import PolarsDataset`. As this knows it is being used in Dataset context it returns row indexes as 1D (what you were looking for) and has a few other niceties.

If you want to roll your own Dataset, you'll need to account for row-wise indexing on DataFrames still being DataFrames, not Series (so you'll need to unpack the extra dimension yourself, eg: `df[0].to_torch()[0]`).

**Comment by alexander-beedie** [comment]
I do have an expansion of the allowed fields in mind (for `PolarsDaraset`, so you aren't limited to "labels" & "features"; I'll see if I can get that done over the Christmas holidays 🤔

As for the CustomDataset, there isn't really anything to look in to; if you index appropriately you'll get the results you want (eg: `df[0].to_torch()[0]`).

Would an enhanced `PolarsDataset` (supporting arbitrary additional features, like "weights", etc) meet your needs? I'll push it up the priority on my backlog if so 👍
